src/app-pages/map/components/map-details/components/dam-profile-chart/DamProfileChart.js

```diff
@@ -0,0 +1,1234 @@
+    .attr("text-anchor", "start")
+    .attr("dx", 250)
+    .attr("dy", damScale(value) + 20)
+    .attr("fill", "#666666")
+    .attr("font-family", "sans-serif")
+    .attr("font-size", "24px")
+    .text(`${value}'`);
+};
+
+const renderDamProfileChart = (data) => {
+  const dpc = d3.select("#dpc-1");
+  const {
+    mode,
+    hasLock,
+    hasTurbine,
+    damTop,
+    damBottom,
+    horizontalLabels,
+    currentLevel,
+    tailWater,
+    inflow,
+    outflow,
+    surcharge,
+    text,
+    date,
+    /* Unused for now but were present in old Dam Profile code so we may need them
+    ruleCurve,
+    precip,
+    designCapacity,
+    gradientTop,
+    gradientBottom,
+    gradientLabel,
+    colorArr,
+    colorLevels,
+    levelType
+    */
+  } = data;
+
+  const tickScale = d3.scaleLinear().domain([0, 17]).range([damTop, damBottom]);
+
+  const damScale = d3.scaleLinear().domain([damTop, damBottom]).range([130, 560]);
+
+  // unused for now - belongs in createMiddleGradient func?
+  // const gradientScale = d3
+  //   .scaleOrdinal()
+  //   .domain([0, 20, 40, 60, 80, 100])
+  //   .range([damScale(gradientTop), damScale(gradientBottom)]);
+
+  // const gradientAxis = d3.axisRight(gradientScale);
+
+  //create line on the left
+  dpc
+    .append("g")
+    .attr("class", "leftLine");
+  dpc
+    .select("g.leftLine")
+    .append("path")
+    .attr(
+      "d",
+      curvedLine([
+        [0, 100],
+        [220, 100],
+        [243, 130],
+        [243, 150],
+        [243, 560],
+        [243, 560],
+        [243, 560],
+      ])
+    )
+    .attr("stroke", "#B3B3B3")
+    .attr("stroke-width", 7)
+    .attr("fill", "none")
+    .attr("stroke-linejoin", "miter");
+  //create center dam
+  dpc
+    .select("g.leftLine")
+    .append("path")
+    .attr(
+      "d",
+      straightLine([
+        [410, 561],
+        [410, 150],
+        [390, 150],
+        [390, 130],
+        [510, 130],
+        [510, 150],
+        [490, 170],
+        [490, 210],
+        [560, 440],
+        [610, 440],
+        [610, 561],
+      ])
+    )
+    .attr("stroke", "#B3B3B3")
+    .attr("stroke-width", 2)
+    .attr("fill", "#B3B3B3")
+    .attr("stroke-linejoin", "bevel");
+
+  drawWaterLevel(dpc, currentLevel, damScale);
+  createLegend(dpc);
+  createInflowIcon(dpc);
+  drawMountain(dpc);
+
+  if (hasLock) {
+    createLock(dpc);
+    drawBoat(dpc);
+  } else {
+    if (hasTurbine) {
+      noLockTurbine(dpc);
+    } else {
+      noLock(dpc);
+    } 
+  }
+  if (hasTurbine) {
+    createTurbine(dpc);
+  }
+
+  drawTicks(dpc, tickScale);
+  createOutflowIcon(dpc, mode);
+  createSurchargeIcon(dpc);
+  createMiddleGradient(dpc, damScale);
+  drawDashedLines(dpc, horizontalLabels, damTop, damBottom, damScale);
+  setText(dpc, mode, inflow, outflow, surcharge, tailWater, text, date);
+};
+
+const DamProfileChart = ({ data }) => {
+  // After SVG is available in the DOM
+  useEffect(() => {
+    renderDamProfileChart( data );
+  }, []);
```

Comment:
@dlin2102 if we remove `data` from the useEffect array, then if you switch between one dam location and another, the d3 svg does not update to the new dam. Also it's not good practice to leave the useEffect array empty if the useEffect relies on a prop variable. I'm still thinking what the proper solution for this would be though

Reply:
Hmm wouldn't it remount every time you switch? Or am I thinking through it wrong?

Reply:
maybe it just needs to keep data in the array, but have the renderDamProfileChart destroy any existing diagram before rebuilding it?

Reply:
It would only remount if the component is removed from the dom and re-added. When you switch between locations the component stays the same and only the prop changes

Reply:
@brian428 I though about that too, but then every time you toggle the accordion or open and close the details view, it causes the DamProfile component to rerender, which in turn would case the DamProfileChart to destroy the svg and rebuild it, even though the dam data itself doesn't change. Though could get inefficient, destroying and rebuilding the chart, every time the accordion or details view is toggled.

Reply:
yep that seems to work.

Reply:
Hmm could we do a name check and then do a check if the elements already exist?

Reply:
Yea that's one solution, I think we can also use useMemo in the DamProfile component so the `data` prop is only recomputed when the locationDetailData changes.

Reply:
I pushed up a commit using useMemo, I think it should be good now

Reply:
@dlin2102 I added the destroy old svg back in. We still need that when switching between two different dam locations so that the previous dam svg will be removed. The good is that now the svg is only being removed when the location detail data changes, not when you toggle the accordion or move around on the map.

Reply:
Cool cool cool. Good job.

Reply:
actually wait. I don't think we need destroy? UseMemo fixed the issue. I need to use that more